src/mlang/backend_compilers/bir_to_dgfip_c.ml

```diff
@@ -581,9 +581,9 @@ let generate_mpp_function (program : Bir.program)
     mppf_stmts
     (if ret_type then
      {|#ifdef FLG_MULTITHREAD
-      return irdata->tas_discord;
+      return irdata->discords;
 #else
-      return tas_discord;
+      return discords;
```

Comment:
And for this one, we weren't returning the correct discord array (tas_discord is used internally to reuse previously allocated discord records, so it points to the set of errors of the previous test case)